docs/flask.md

```diff
@@ -0,0 +1,202 @@
+project initialization, it did not support Python 3.8 yet, so we are using
+Python 3.7.
+
+Using [Docker](https://github.com/ccmbioinfo/stager/blob/master/docs/docker.md)
+will automatically set everything up for you and works well in general. For some
+specific debugging and database purposes, you may need to set up a virtualenv.
+
+## Setting up in a virtualenv
+
+First, you will need Python 3.7, pip, and virtualenv. To create a virtualenv,
+switch to the `flask` directory and run one of the following commands,
+depending on how your Python installation is set up.
+
+A virtualenv changes your PATH so Python modules resolve to dependencies
+installed in this virtual environment, which helps keep projects separated
+instead of polluting your global site packages.
+
+```bash
+virtualenv venv
+# if the above doesn't work
+python3 -m virtualenv
+```
+
+Then install dependencies through pip in the virtualenv.
+
+```bash
+. venv/bin/activate  # Activates the virtualenv. Run in a shell before dev work
+pip3 install -r requirements-dev.txt
+```
+
+Install the [MinIO command-line client](https://docs.min.io/docs/minio-client-quickstart-guide.html)
+and make sure that it is in PATH.
+
+Assuming the supporting MySQL database server and MinIO server are up and access
+credentials are in the environment variables, run
+
+```bash
+./utils/run.sh
+```
+
+Additional parameters are passed to Flask, so you can do
+`./utils/run.sh --host=0.0.0.0` to make your development server accessible
+externally (insecure). Set FLASK_ENV=development to watch files and restart
+automatically, and FLASK_DEBUG=1 to enable debugging.
+
+If you are using [PyCharm](https://github.com/ccmbioinfo/stager/blob/master/docs/pycharm.md)
+the Flask server run configuration takes care of this for you, and you don't
+need the startup helper script. The module it should use is `wsgi`.
+
+The run script supports setting environment variables PYTHON and COMMAND to
+choose your Python interpreter. By default, it runs `python3 -m flask`, but you
+can set `COMMAND=flask` to have it call `flask` directly, or `PYTHON=python3.7`
+to have it run `python3.7 -m flask`, for example.
+
+## Adding or updating dependencies
+
+Stager uses [pip-tools](https://github.com/jazzband/pip-tools) to manage
+`requirements.txt` for production and `requirements-dev.txt` in development.
+It needs to be installed separately and is not included in the Docker dev image
+for now.
+
+To add a runtime dependency, add a line specifying the name of the PyPI package
+to `requirements.in`. Keep in mind to run these in a virtualenv if you are using
+one!
+
+```bash
+pip-compile requirements.in
+pip install -r requirements.txt
+```
+
+To add a development dependency, add the line to `requirements-dev.in` instead.
+
+```bash
+pip-compile requirements-dev.in
+pip install -r requirements-dev.in
+```
+
+Commit the changes to Git. If you are using Docker, you should rebuild the image
+for changes to persist.
```

Comment:
I think it would be helpful to add the command (`docker-compose up --build` )for rebuilding the docker image here since this has come up a few times in slack after a new python dependency is introduced

Reply:
Yeah, I'll do that instead of assuming a reference to the Docker part of the documentation that I wrote first.